File: app/src/main/java/com/jeksonshar/jweather/repository/room/WeatherDao.java
package com.jeksonshar.jweather.repository.room;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface WeatherDao {

    @Query("SELECT*FROM WeatherEntity")
    List<WeatherEntity> getAllItemsWeather();

    @Query("SELECT*FROM WeatherEntity WHERE id == :id")
    WeatherEntity getByID(String id);

    @Insert
    void saveWeatherModel(WeatherEntity weatherEntity);

    @Update
    void updateWeatherModel(WeatherEntity entity);
}

File: app/src/main/java/com/jeksonshar/jweather/repository/room/WeatherDao.kt
package com.jeksonshar.jweather.repository.room

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

@Dao
interface WeatherDao {
    @get:Query("SELECT*FROM WeatherEntity")
    val allItemsWeather: List<WeatherEntity?>? //TODO лучше делать консистентно, два разных подхода в одном классе

    @Query("SELECT*FROM WeatherEntity WHERE id == :id")
    fun getByID(id: String?): WeatherEntity?

    @Insert
    fun saveWeatherModel(weatherEntity: WeatherEntity?)

    @Update
    fun updateWeatherModel(weatherEntity: WeatherEntity?)
}